pkg/codegen/go/gen_test.go

```diff
@@ -166,6 +168,44 @@ func readSchemaFile(file string) *schema.Package {
 	return pkg
 }
 
+func readYamlSchemaFile(file string) *schema.Package {
+	// Read in, decode, and import the schema.
+	schemaBytes, err := os.ReadFile(filepath.Join("..", "testing", "test", "testdata", file))
+	if err != nil {
+		panic(err)
+	}
+	var pkgSpec schema.PackageSpec
+	if err = yaml.Unmarshal(schemaBytes, &pkgSpec); err != nil {
+		panic(err)
+	}
+	pkg, err := schema.ImportSpec(pkgSpec, map[string]schema.Language{"go": Importer})
+	if err != nil {
+		panic(err)
+	}
+
+	return pkg
+}
+
+func TestLanguageResources(t *testing.T) {
+	t.Parallel()
+
+	for _, test := range test.PulumiPulumiSDKTests {
+		test := test
+		t.Run(test.Directory, func(t *testing.T) {
+			t.Parallel()
+			var pkg *schema.Package
+			if test.Directory == "simple-yaml-schema" || test.Directory == "cyclic-types" {
+				pkg = readYamlSchemaFile(filepath.Join(test.Directory, "schema.yaml"))
+			} else {
+				pkg = readSchemaFile(filepath.Join(test.Directory, "schema.json"))
+			}
+
+			_, err := LanguageResources("test", pkg)
```

Comment:
Would it be worth validating what we're getting back here too?  It would be a little bit more tricky, but would also make the test more significant I think?

Reply:
Yes this should at least check the names of things match, probably brittle to test the entire structure.

Reply:
Added an assert to verify that the name of the resource is derived from the token ✅

Reply:
But isn't the whole point of LanguageResources that the name _isn't_ derived from the token?

Reply:
All implementations of `LanguageResources(...)` derive the name of the resource from the token and they all use `tokenToName(...)` so I am not sure what the point should be

Reply:
All implementations of `LanguageResources(...)` derive the name of the resource from the token and they all use `tokenToName(...)` so I am not sure what the point should be